client/asset/eth/node.go

```diff
@@ -219,22 +228,41 @@ func runNode(cfg *nodeConfig) (*node.Node, error) {
 	case dex.Mainnet:
 		// urls = params.MainnetBootnodes
 		// TODO: Allow.
-		return nil, fmt.Errorf("eth cannot be used on mainnet")
+		return fmt.Errorf("eth cannot be used on mainnet")
 	default:
-		return nil, fmt.Errorf("unknown network ID: %d", uint8(cfg.net))
+		return fmt.Errorf("unknown network ID: %d", uint8(network))
 	}
 
 	ethCfg.SyncMode = downloader.LightSync
 
-	if _, err := les.New(stack, &ethCfg); err != nil {
-		return nil, err
+	if _, err := les.New(node, &ethCfg); err != nil {
+		return err
 	}
 
-	if err := stack.Start(); err != nil {
-		return nil, err
+	if err := node.Start(); err != nil {
+		return err
 	}
 
-	return stack, nil
+	return nil
+}
+
+// importAccountToNode imports an account into the ethereum wallet by private key
+// that can be unlocked with password.
+func importAccountToNode(node *node.Node, privateKey, password []byte) error {
+	ecdsaPrivateKey, err := crypto.ToECDSA(privateKey)
+	if err != nil {
+		return err
+	}
+	ks := node.AccountManager().Backends(keystore.KeyStoreType)[0].(*keystore.KeyStore)
+	_, err = ks.ImportECDSA(ecdsaPrivateKey, hex.EncodeToString(password))
```

Comment:
Since `NewWallet` errors if there is more than one account, seems like `importAccountToNode` should refuse to import if there is already one?
Besides dex not using more than one account, was there a consideration for `NewWallet` erroring for `len != 1` instead of just `len <  1` and using the first one in the slice regardless of its length?

Reply:
Yeah `importAccountToNode` could definitely fail to import. Maybe it could even check if the one that's already imported is the same as the one it's attempting to import, and if not, replace it.

I did consider that, but if there are more than one accounts stored, it means something went wrong somewhere. In `NewWallet` we don't have access to the wallet seed, so there's no way make sure that the account that is being loaded is the correct one. This situation should never happen, but maybe in this case we should give the user the option to recreate the wallet?